src/vmspawn/vmspawn.c

```diff
@@ -540,6 +551,10 @@ static int run_virtual_machine(void) {
         if (!machine)
                 return log_oom();
 
+
+        if (arg_kernel && access(arg_kernel, F_OK) < 0)
+                return log_error_errno(errno, "Kernel not found at %s: %m", arg_kernel);
```

Comment:
i don't think we need to do this check here. we'll notice soon enough, no? qemu will fail if it cannot find the path

Reply:
Fixed in eae47ee9d9.